### PR TITLE
Replace manually managed TLS slots with regular thread_local statics

### PR DESCRIPTION
**Pull request by jkotas**
The manually managed TLS slots were workaround for lack of thread-local statics on old Windows versions. This is not the case anymore.

### CODE REVIEW

File: src/coreclr/src/inc/clrinternal.idl [hunk 1]
```diff
@@ -128,30 +128,6 @@ typedef VOID (__stdcall *PTLS_CALLBACK_FUNCTION)(PVOID);
 ]
 interface IExecutionEngine : IUnknown
```
Comment (jkotas) [inline src/coreclr/src/inc/clrinternal.idl]: My ultimate goal is to delete the whole IExecutionEngine indirection. The TLS slots are big enough piece to deserve dedicated PR.

**Comment by jkotas** [comment]
The CI failures are #33596

**Comment by stephentoub** [comment]
Love all the deletion.  Is there any perf impact here, positive or negative?

**Comment by jkotas** [comment]
None of these thread-statics are used on performance critical paths, so the observable effect is noise.

The performance effect should be positive:
- Better throughput: This deletes unnecessary level of indirection. 
- Better footprint: Before this change, we have always allocated fixed number of pointer-sized slots per thread. With this change, the slots are of the right size (e.g. bool instead of pointer) and the debug-only slots are present in debug builds only.

File: src/coreclr/src/vm/util.cpp [hunk 2]
```diff
@@ -21,6 +21,43 @@
...
+
+thread_local size_t t_ThreadType;
+
+void ClrFlsSetThreadType(TlsThreadTypeFlag flag)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    t_ThreadType |= flag;
+
+    // The historic location of ThreadType slot kept for compatibility with SOS
```
Comment (jkotas) [inline src/coreclr/src/vm/util.cpp]: I have verified that SOS prints the names of the special threads correctly.

File: src/coreclr/src/vm/threads.h [hunk 3]
```diff
@@ -950,12 +950,12 @@ typedef DWORD (*AppropriateWaitFunc) (void *args, DWORD timeout, DWORD option);
 // of that physical thread.
 
 // FEATURE_MULTIREG_RETURN is set for platforms where a struct return value
-// [GcInfo v2 only]        can be returned in multiple registers
+//                ]        can be returned in multiple registers
```
Comment (janvorli) [inline src/coreclr/src/vm/threads.h]: A nit - a leftover closing brace

**Review by janvorli — APPROVED** [review]
LGTM, nice cleanup!
